logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -604,7 +605,7 @@ class TestPipeline < LogStash::Pipeline
 
       Thread.new { subject.run }
       # make sure we have received all the generated events
-      sleep 1 while dummyoutput.events.size < number_of_events
+      sleep 0.25 until dummyoutput.events.size == number_of_events
```

Comment:
you are correct the sampling my well miss the point at which the two equate - I should not assume that events.size will never be greater than number_of_events.